pkg/miscellaneous/miscellaneous.go

```diff
@@ -61,6 +61,100 @@ func GetLatestFullSnapshotAndDeltaSnapList(store brtypes.SnapStore) (*brtypes.Sn
 	return fullSnapshot, deltaSnapList, nil
 }
 
+// GetAllBackups returns all backups (full snapshots and all their delta snapshots).
+func GetAllBackups(store brtypes.SnapStore) (brtypes.SnapList, error) {
+	snapList, err := store.List()
+	if err != nil {
+		return nil, err
+	}
+
+	snapshots := brtypes.SnapList{}
+	for i := len(snapList) - 1; i >= 0; i-- {
+		if !snapList[i].IsChunk {
+			snapshots = append(snapshots, snapList[i])
+		}
+	}
+
+	return snapshots, nil
+}
+
+// GetFilteredBackups returns backups (full snapshots and all their delta snapshots) based on the filter function.
+func GetFilteredBackups(store brtypes.SnapStore, filter func(snaps brtypes.SnapList) (brtypes.SnapList, error)) (brtypes.SnapList, error) {
```

Comment:
I think there are too many functions that perform the same task with slight variations. I would combine all functions starting with `GetAllBackups` and ending with `FilterLastSnapshotsAndItsDeltasSinceNDays` with a single function `GetFilteredSnapshots` that accepts a store, a max number of backups (0 for all), and a "predicate" function to filter a single full snapshot (nil for no effect), and that performs all fetching, sorting, grouping, and filtering.
It should return `(brtypes.SnapList, error)` and therefore the notion of `backup` would be internal to it and not leak into any external names.